src/components/Cluster/ClusterDetail/ClusterDetail.js

```diff
@@ -11,37 +11,47 @@ import { AppRoutes } from 'shared/constants/routes';
 import GettingStarted from '../../GettingStarted/GettingStarted';
 import ClusterDetailView from './ClusterDetailView';
 
-const ClusterDetail = (props) => {
+const ClusterDetail = ({ match, cluster, clusterId, ...rest }) => {
   const dispatch = useDispatch();
 
-  if (!props.cluster) {
-    new FlashMessage(
-      `Cluster <code>${props.clusterId}</code> doesn't exist.`,
-      messageType.INFO,
-      messageTTL.MEDIUM
-    );
+  useEffect(() => {
+    if (!cluster) {
+      new FlashMessage(
+        `Cluster <code>${clusterId}</code> doesn't exist.`,
+        messageType.INFO,
+        messageTTL.MEDIUM
+      );
 
-    dispatch(push(AppRoutes.Home));
+      dispatch(push(AppRoutes.Home));
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
```

Comment:
Only executing this effect on component mount, to not interfere with the existent logic in `ClusterDetailView`